modules/tinymce/src/plugins/link/main/ts/core/Actions.ts

```diff
@@ -50,7 +50,7 @@ const setupGotoLinks = (editor: Editor): void => {
 
   editor.on('keydown', (e) => {
     const link = getSelectedLink(editor);
-    if (link && e.keyCode === 13 && hasOnlyAltModifier(e)) {
+    if (!e.defaultPrevented && link && e.keyCode === 13 && hasOnlyAltModifier(e)) {
```

Comment:
This should be using the `e.isDefaultPrevented()` function instead, as that's the TinyMCE provided API that is designed to handle cross browser cases and custom events.

Additionally, this isn't the best place to put this. Currently, you're running `getSelectedLink()` when it's never going to be used, so this check should instead be around the entire block instead of just this if statement.

Reply:
I noticed this as well that the more modern property was used however since we are dropping IE support we could start using this property. This wasn't a standard property at the time so the naming here of the function of the workaround is more based on how it works in other popular frameworks at the time. But I guess for custom events it would be wrong since it will not set the `defaultPrevented` flag if you can `preventDefault` on those but maybe it should then we could slowly migrate toward the more standard property. So for now change it but should we consider a modernization here?

Reply:
Is afk so couldn't test this it seems that the normalization code for events already handle the property state change. https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/core/main/ts/events/EventUtils.ts#L86

Reply:
> Additionally, this isn't the best place to put this. Currently, you're running `getSelectedLink()` when it's never going to be used, so this check should instead be around the entire block instead of just this if statement.

@lnewson   It seems the result of `getSelectedLink()` is used in the if condition. Should I remove it from the if condition and put it inside the body of if statement ?

Reply:
I'm not sure how that relates to what I mentioned sorry? To be clear, this is what I expected to see here:
```
  editor.on('keydown', (e) => {
    if (!e.isDefaultPrevented()) {
      const link = getSelectedLink(editor);
      if (link && e.keyCode === 13 && hasOnlyAltModifier(e)) {
        e.preventDefault();
        gotoLink(editor, link);
      }
    }
  });
```
In saying that though, now that we have a separate if condition we can optimize the above a little more, such as:
```
  editor.on('keydown', (e) => {
    if (!e.isDefaultPrevented() && e.keyCode === 13 && hasOnlyAltModifier(e)) {
      const link = getSelectedLink(editor);
      if (link) {
        e.preventDefault();
        gotoLink(editor, link);
      }
    }
  });
```
This way we're only doing the link lookup if alt+enter is pressed and the event hasn't been prevented. Ideally we'd also use a non falsy check for `link` as well.

Reply:
Hey @lnewson should this conversation be resolved ?

Reply:
Yep, it's all good now. The point Spocke mentioned I've been talking to him via DM's about that and we're going to raise it via #tinymce-dev.